File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
    Arrays.sort(phone_book);
        for(int j=0; j<phone_book.length-1; j++){
            if(phone_book[j].length() > phone_book[j+1].length()){
                if(phone_book[j].substring(0,phone_book[j+1].length()).equals(phone_book[j+1])){
                    return false;
                }
            }
            else{
                if(phone_book[j+1].substring(0,phone_book[j].length()).equals(phone_book[j])){
                    return false;
                }
            }
        }
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort()
    for i in range(len(phone_book)-1):
        for j in range(i+1, len(phone_book)):
            if len(phone_book[i]) < len(phone_book[j]) and phone_book[i][0] == phone_book[j][0]:
                if phone_book[j][:len(phone_book[i])] == phone_book[i]:
                    return False
            else:
                break
    
    return True
            